pkg/perf/perf_suite_test.go

```diff
@@ -0,0 +1,154 @@
+	serverConfig                *ServerConfig
+	measurementConfig           *MeasurementConfig
+	measurementConfigPath       string
+	singleImagePushTime         int
+	parallelPushTime            int
+	singleImageParallelPushTime int
+	times                       int
+)
+
+// nolint: gochecknoinits
+func init() {
+	flag.StringVar(&serverConfigPath, "server.config", "",
+		"path to the server config file")
+	flag.StringVar(&measurementConfigPath, "measurement.config", "",
+		"path to the measurement config file")
+}
+
+func TestPerf(t *testing.T) {
+	t.Skip("Skip running!")
+
+	RegisterFailHandler(Fail)
+
+	Expect(serverConfigPath).To(BeAnExistingFile(),
+		"Invalid test suite argument. server.config should be an existing file.")
+
+	serverConfig = LoadServerConfig(serverConfigPath)
+	op = skopeoOp{
+		serverConfig.Username,
+		serverConfig.Password,
+		serverConfig.Address,
+		serverConfig.TLSVerify,
+		serverConfig.Repo,
+	}
+
+	Expect(measurementConfigPath).To(BeAnExistingFile(),
+		"Invalid test suite argument. measurement.config should be an existing file.")
+
+	measurementConfig = LoadMeasurementConfig(measurementConfigPath)
+	singleImagePushTime = measurementConfig.SingleImagePushTime
+	parallelPushTime = measurementConfig.ParallelPushTime
+	singleImageParallelPushTime = measurementConfig.SingleImageParallelPushTime
+	times = measurementConfig.Times
+
+	// Pushing images for pull them later
+	imageName := "zot-tests-dummy-pull"
+	Expect(runPushCommand(op, imageName)).To(Equal(true))
+
+	RunSpecs(t, "Performance Suite")
+}
+
+var _ = BeforeSuite(func() {
+	err := os.MkdirAll("pulled_images", 0777)
+	if err != nil {
+		log.Fatalln("Error creating pulled_images dir!", err)
+	}
+})
+
+var _ = AfterSuite(func() {
+	// err := os.RemoveAll("pulled_images")
+	// if err != nil {
+	// 	log.Fatalln("Error removing pulled_images dir!", err)
+	// }
+})
+
+var _ = Describe("Check Zot Performance", func() {
+	Measure("Performance measuring - push", func(b Benchmarker) {
+		runtime := b.Time("runtime", func() {
+			imageName := "zot-tests-dummy-push"
+			Expect(runPushCommand(op, imageName)).To(Equal(true))
+			Expect(runDeleteCommand(op, imageName)).To(Equal(true))
+		})
+		Expect(runtime.Seconds()).To(BeNumerically("<", singleImagePushTime),
+			"Push oprations shouldn't take too long!")
+	}, times)
+
+	Measure("Performance measuring - parallel push", func(b Benchmarker) {
+		runtime := b.Time("runtime", func() {
+			imageNameIdx := "zot-tests-parallel-images-dummy-%d"
+			var commands []string
+			var pushedImages []string
+
+			for i := 1; i <= 5; i++ {
+				imageName := fmt.Sprintf(imageNameIdx, i)
+				pushedImages = append(pushedImages, imageName)
+				arguments := setCopyCommand(op)
+
+				arguments = setCommandVariables(op, arguments, imageName, imageName, true)
+				commands = append(commands, strings.Join(arguments, " "))
+			}
+
+			Expect(runCommands(commands)).To(Equal(true))
+
+			for _, imageName := range pushedImages {
```

Comment:
I see here you are removing images, but not in the rest of tests.

Maybe we should remove all of them after the stress test?

Reply:
This is my intention, to remove the images at the end of the tests. Apparently I missed this on one of the pushing tests.

Reply:
ginkgo also does a Before..() and After...() which may be cleaner to read.

Global Setup and Teardown: BeforeSuite and AfterSuite
https://onsi.github.io/ginkgo/